app/services/setting_registry.rb

```diff
@@ -0,0 +1,100 @@
+class SettingRegistry
+  include Singleton
+
+  def initialize
+    @settings = {}
+    load
+  end
+
+  def logger
+    Foreman::Logging.logger('app')
+  end
+
+  def find(name)
+    @settings[name.to_s]
+  end
+
+  # Returns a setting effective value
+  # Call as `Foreman.settings[<name>]`
+  #
+  #   Foreman.settings['default_locale'] => nil.
+  #   Foreman.settings[:default_locale] => nil.
+  #
+  def [](name)
+    definition = find(name)
+    logger.warn("Setting #{name} has no definition, please define it before using") and return unless definition
+    definition.value
+  end
+
+  def []=(name, value)
+    definition = find(name)
+    raise ::Foreman::Exception.new(N_("Setting definition for '%s' not found, can not set"), name) unless definition
+    db_record = _find_or_new_db_record(name)
+    db_record.value = value
+    db_record.save!
+    definition.value = db_record.value
+  end
+
+  def load
+    # add() all setting definitions
+    Setting.descendants.each do |cat_cls|
```

Comment:
Looks like when starting the app this runs before core settings are loaded (meaning the `Setting.descendants` doesn't include all the subclassess yet), because we extend ActiveRecord with a method that calls `Setting[:per_page]` which triggers the singleton initializer:
https://github.com/theforeman/foreman/blob/develop/lib/core_extensions.rb#L119

Which in turn leads to logging a lot of messages, meaning we are using settings before the registery is fully loaded and initialized.

Reply:
yes we are and it's known issue I believe, but there is not much we can do about it in this PR, in follow-up I think we should improve that and try not accessing settings too early.

Reply:
I believe I've 'fixed' it, by manual initialization and by throwing a warning if setting is used too soon